wasm/jit/engine.go

```diff
@@ -420,17 +422,23 @@ func (e *engine) exec(f *compiledWasmFunction) {
 	}
 }
 
-func (e *engine) memoryGrow(m *wasm.MemoryInstance, newPages uint64) {
+func (e *engine) builtinFunctionMemoryGrow(f *compiledWasmFunction) {
+	newPages := e.pop()
 	max := uint64(math.MaxUint32)
-	if m.Max != nil {
-		max = uint64(*m.Max) * wasm.PageSize
+	if f.memory.Max != nil {
+		max = uint64(*f.memory.Max) * wasm.PageSize
 	}
 	// If exceeds the max of memory size, we push -1 according to the spec
-	if uint64(newPages*wasm.PageSize+uint64(len(m.Buffer))) > max {
+	if uint64(newPages*wasm.PageSize+uint64(len(f.memory.Buffer))) > max {
 		v := int32(-1)
 		e.push(uint64(v))
 	} else {
-		e.push(uint64(uint64(len(m.Buffer)) / wasm.PageSize))
-		m.Buffer = append(m.Buffer, make([]byte, newPages*wasm.PageSize)...)
+		e.push(uint64(uint64(len(f.memory.Buffer)) / wasm.PageSize))
```

Comment:
nit: re-use `f.builtinFunctionMemorySize() // grow returns the prior memory size on change`

Reply:
https://github.com/tetratelabs/wazero/pull/85/commits/1b6114f43181c5f4c39e34e54a9c0cb86c90ab4f